utils/signAndSendTx.ts

```diff
@@ -0,0 +1,59 @@
+import { Api } from "@cennznet/api";
+import { Signer, SubmittableExtrinsic } from "@cennznet/api/types";
+
+interface TxReceipt {
+	hash: string;
+	events: any[];
+}
+
+export default async function signAndSendTx(
+	api: Api,
+	extrinsic: SubmittableExtrinsic<"promise", any>,
+	address: string,
+	signer: Signer
+): Promise<TxReceipt | "cancelled"> {
+	const signAndSend = async () => {
+		return new Promise<TxReceipt>((resolve, reject) => {
+			extrinsic
+				.signAndSend(address, { signer }, (progress) => {
+					const { dispatchError, status, events } = progress;
+					if (dispatchError && dispatchError?.isModule && status.isFinalized) {
+						const { index, error } = dispatchError.asModule.toJSON();
+						const errorMeta = api.registry.findMetaError(
+							new Uint8Array([index, error])
+						);
+						const errorCode =
+							errorMeta?.section && errorMeta?.name
+								? `${errorMeta.section}.${errorMeta.name}`
+								: `I${index}E${error}`;
+
+						return reject(
+							new Error(`${errorCode}:${status?.asFinalized?.toString()}`)
+						);
+					}
+					if (status.isFinalized)
```

Comment:
would be nice, if we wait till isInBlock to resolve the promise
It will still give you the required txHash and events..